ingenialink/ethernet/network.py

```diff
@@ -41,32 +43,36 @@ class NetStatusListener(Thread):
     def __init__(self, network):
         super(NetStatusListener, self).__init__()
         self.__network = network
-        self.__state = NET_STATE.CONNECTED
         self.__stop = False
         self.__max_unsuccessful_pings = MAX_NUM_UNSUCCESSFUL_PINGS
 
     def run(self):
-        servo = self.__network.servos[0]
-        unsuccessful_pings = 0
         while not self.__stop:
-            response = ping(servo.ip_address, timeout=1)
-            if not isinstance(response, float):
-                unsuccessful_pings += 1
-            else:
-                unsuccessful_pings -= 1
-                unsuccessful_pings = max(0, unsuccessful_pings)
-            if unsuccessful_pings > self.__max_unsuccessful_pings:
-                if self.__state != NET_STATE.DISCONNECTED:
-                    self.__state = NET_STATE.DISCONNECTED
-                    self.__network.status = NET_STATE.DISCONNECTED
-                    self.__network._notify_status(NET_DEV_EVT.REMOVED)
-                unsuccessful_pings = self.__max_unsuccessful_pings
-            elif unsuccessful_pings == 0:
-                if self.__state != NET_STATE.CONNECTED:
-                    self.__network.status = NET_STATE.CONNECTED
-                    self.__state = NET_STATE.CONNECTED
-                    self.__network._notify_status(NET_DEV_EVT.ADDED)
-            sleep(1)
+            for servo in self.__network.servos:
+                unsuccessful_pings = 0
+                servo_ip = servo.ip_address
+                servo_state = self.__network._get_servo_state(servo_ip)
+                while unsuccessful_pings < self.__max_unsuccessful_pings:
+                    response = ping(servo_ip, timeout=1)
+                    if not isinstance(response, float):
+                        unsuccessful_pings += 1
+                    else:
+                        break
+                ping_response = (unsuccessful_pings !=
+                                 self.__max_unsuccessful_pings)
+                if servo_state == NET_STATE.CONNECTED:
+                    if ping_response:
+                        time.sleep(0.25)
+                    else:
+                        servo_state = NET_STATE.DISCONNECTED
+                        self.__network._notify_status(servo_ip,
+                                                      NET_DEV_EVT.REMOVED)
+                elif ping_response:
+                    servo_state = NET_STATE.CONNECTED
+                    self.__network._notify_status(servo_ip,
+                                                  NET_DEV_EVT.ADDED)
```

Comment:
I think you can simplify the if:
```python
if servo_state == NET_STATE.CONNECTED and not ping_response:
    self.__network._notify_status(servo_ip, NET_DEV_EVT.REMOVED)
    self.__network._set_servo_state(servo_ip, NET_STATE.DISCONNECTED)
if servo_state == NET_STATE.DISCONNECTED and ping_response:
    self.__network._notify_status(servo_ip, NET_DEV_EVT.ADDED)
    self.__network._set_servo_state(servo_ip, NET_STATE.CONNECTED)


 ```

Reply:
This does not contemplate the sleep function.

Reply:
Done